src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetCompilePdfHelper.java

```diff
@@ -442,7 +447,23 @@ private String detectRootDirective(ArrayList<TexMagicComment> magicComments)
       return null;
    }
    
+   private void showWarning(WarningBarDisplay display, String message)
+   {
+      isWarningShowing_ = true;
+      display.showWarningBar(message);
+   }
+   
+   private void hideWarning(WarningBarDisplay display)
+   {
+      if (isWarningShowing_)
```

Comment:
This flag is used to make sure this code doesn't unintentionally hide a warning bar displayed from a separate component in the IDE (e.g. the debug highlight).

Really, we probably need to update the API here to "add" and "remove" warnings, with each warning having some unique identifier, so there's a way for the view to handle multiple concurrent warnings in some sensible way... but that's for another day.